client/src/components/Commits.js

```diff
@@ -22,10 +22,41 @@ class Commits extends React.Component {
     return obj;
    })
    .filter(item => item.sourcestamps[0].revision === commit.sha);
+  let BSID = filteredBs.map(id => {
```

Comment:
Commits component is a presentational one, it shouldn't have that complicated logic. Builds data should be put into `commit` by something else (so that Commits component just takes an array of `commits` from the state and they already have builds data).

This code should be separated between reducers and sagas

Reply:
of course, I was just playing with filter, will remove this in a later commit for sure